### ISSUE TITLE
refactor/rename feature not working

### ISSUE THREAD

**Issue by bend-n** [godotengine/godot-vscode-plugin#361]
### Godot version

3.4

### VS Code version

1.66.2

### Godot Tools VS Code extension version

v1.3.0

### System information

ArchLinux x11 

### Issue description

Has issues with differentiating between functions and variables.

https://user-images.githubusercontent.com/70787919/168254686-61f5b27f-6664-4e70-bef7-6e8b3cb90024.mp4

### Steps to reproduce

```gdscript
extends Node

var three = "threes"

func three():
    return
```

**Comment by Razoric480** [comment]
Interesting. The LSP should be checking that the two symbols are the exact same, but clearly there's some things the parser isn't doing.

**Comment by atirut-w** [comment]
Is there already an open issue for this in [godotengine/godot](https://github.com/godotengine/godot)?

**Comment by atirut-w** [comment]
This is a pretty big issue for any script that are remotely complex and it is still happening. There does not seem to be any open issue for this on the Godot repo either.

It's been *months*...

**Comment by DaelonSuzuka** [comment]
> There does not seem to be any open issue for this on the Godot repo either.

![giphy 1](https://user-images.githubusercontent.com/18042232/189301592-b362c461-4cb3-4e1b-bfab-0e57263e385b.gif)

> It's been *months*...

![giphy 1](https://user-images.githubusercontent.com/18042232/189301928-ac7842da-0b64-4bd4-a81d-f518ebf3c984.gif)

Unfortunately, this just isn't fixable from the extension side. As you can see in the capture below, the language server simply returns `"changes": {}` when you send a rename command. 

```
[godot-tools][2022-09-09T07:41:29.820Z] [client] {"jsonrpc":"2.0","id":15,"method":"textDocument/rename","params":{"textDocument":{"uri":"file:///p%3A/isotope/isotope_project/player/Player.gd"},"position":{"line":4,"character":17},"newName":"inventory2"}}
[godot-tools][2022-09-09T07:41:30.336Z] [server] {"id":15,"jsonrpc":"2.0","result":{"changes":{}}} 
```

I also tried in a Godot 4 project, and got the same result:

```
[godot-tools][2022-09-09T08:31:38.048Z] [client] {"jsonrpc":"2.0","id":6,"method":"textDocument/rename","params":{"textDocument":{"uri":"file:///p%3A/_godot/SelectionShortcuts2/addons/selectionshortcuts2/plugin.gd"},"position":{"line":101,"character":10},"newName":"target2"}} 
[godot-tools][2022-09-09T08:31:38.117Z] [server] {"id":6,"jsonrpc":"2.0","result":{"changes":{}}} 
```

**Comment by bend-n** [comment]
Can we create a https://godotengine/godot issue?

**Comment by atirut-w** [comment]
Done

**Comment by DaelonSuzuka** [comment]
Renames have been implemented in the language server as of Godot 4.2.
